framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -0,0 +1,718 @@
+    self.raiseADebug(self.convFormat.format(name='temperature',
+                                            conv=str(converged),
+                                            got=self.T,
+                                            req=self._convergenceCriteria['temperature']))
+    return converged
+
+  def _checkForImprovement(self, new, old):
+    """
+      Determine if the new value is sufficient improved over the old.
+      @ In, new, float, new optimization value
+      @ In, old, float, previous optimization value
+      @ Out, improved, bool, True if "sufficiently" improved or False if not.
+    """
+    # This is not required for simulated annealing as it's handled in the probabilistic acceptance criteria
+    # But since it is an abstract method it has to exist
+    pass
+
+  def _checkAcceptability(self, traj, opt, optVal, info):
+    """
+      Check if new opt point is acceptably better than the old one
+      @ In, traj, int, identifier
+      @ In, opt, dict, new opt point
+      @ In, optVal, float, new optimization value
+      @ Out, acceptable, str, acceptability condition for point
+      @ Out, old, dict, old opt point
+    """
+    # Check acceptability
+    # NOTE: if self._optPointHistory[traj]: -> faster to use "try" for all but the first time
+    try:
+      old, _ = self._optPointHistory[traj][-1]
+      oldVal = old[self._objectiveVar]
+      # check if same point
+      self.raiseADebug(' ... change: {d: 1.3e} new objective: {n: 1.6e} old objective: {o: 1.6e}'
+                      .format(d=opt[self._objectiveVar]-oldVal, o=oldVal, n=opt[self._objectiveVar]))
+      # if this is an opt point rerun, accept it without checking.
+      if self._acceptRerun[traj]:
+        acceptable = 'rerun'
+        self._acceptRerun[traj] = False
+      elif all(opt[var] == old[var] for var in self.toBeSampled):
+        # this is the classic "same point" trap; we accept the same point, and check convergence later
+        acceptable = 'accepted'
+      else:
+        if self._acceptabilityCriterion(oldVal,opt[self._objectiveVar])>randomUtils.random(dim=1, samples=1): # TODO replace it back
+          acceptable = 'accepted'
+        else:
+          acceptable = 'rejected'
+    except IndexError:
+      # if first sample, simply assume it's better!
+      acceptable = 'first'
+      old = None
+    self._acceptHistory[traj].append(acceptable)
+    self.raiseADebug(' ... {a}!'.format(a=acceptable))
+    return acceptable, old
+
+  def _acceptabilityCriterion(self,currentObjective,newObjective):
+    """
+      Check if new opt point is acceptably better than the old one
+      @ In, currentObjective, float, the current value of the objective function (i.e., current energy)
+      @ In, newObjective, float, the value of the objective function at the new candidate
+      @ Out, Prob, float, the acceptance probability
+    """
+    # Boltzman Constant
+    kB = 1 #1.380657799e-23 # or 1
+    if self.T0 == None:
+      self.T0 = 1e4
```

Comment:
this should be handled at the input file reading stage..not here

Reply:
T0 is not a user input.

Reply:
Okay. So it should be set in the __init__  method. Not in here.